avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -1445,7 +1446,7 @@ private Schema toConnectSchema(org.apache.avro.Schema schema, boolean forceOptio
       while (paramIt.hasNext()) {
         Map.Entry<String, JsonNode> field = paramIt.next();
         JsonNode jsonValue = field.getValue();
-        if (!jsonValue.isTextual()) {
+        if ((!field.getKey().endsWith(DOC_PROP) && !jsonValue.isTextual())) {
```

Comment:
@ewencp I agree. There might be other cases where missing (non-mandatory) fields cause a connector to fail for a given schema. 

In addition, the workaround (in case of the HDFS sink connector) is very invasive:
- Delete connector
- Evolve schema to add missing "doc" field
- Restart producer application in order to use the new schema
- Delete existing messages from topic (by either deleting the topic or by setting retention time to a small value and waiting)
- Delete existing files from HDFS
- Create connector

In essence, this causes data loss (both in Kafka and HDFS) and downtime of the producer system, which is the worst case scenario in production. A fix would be much appreciated.